.github/workflows/build.yml

```diff
@@ -30,16 +30,12 @@ jobs:
       # on Maven Central, and once against the Picnic Error Prone fork,
       # additionally enabling all checks defined in this project and any Error
       # Prone checks available only from other artifact repositories.
-      - name: Check out code
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-        with:
-          persist-credentials: false
-      - name: Set up JDK
-        uses: actions/setup-java@387ac29b308b003ca37ba93a6cab5eb57c8f5f93 # v4.0.0
+      - name: Check out code and set up JDK and Maven
+        uses: s4u/setup-maven-action@fa2c7e4517ed008b1f73e7e0195a9eecf5582cd4 # v1.11.0
         with:
           java-version: ${{ matrix.jdk }}
-          distribution: ${{ matrix.distribution }}
-          cache: maven
```

Comment:
Why did we drop the cache ?

Did not find [anything](https://www.google.com/search?q=codeql+analysiss+and+cache&rlz=1C5CHFA_enNL1017NL1017&oq=codeql+analysiss+and+cache&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIJCAEQIRgKGKAB0gEINDc1OWowajeoAgCwAgA&sourceid=chrome&ie=UTF-8) that says this is enabled by default 🤔

Reply:
Good question. The implementation shows that the check [includes](https://github.com/s4u/setup-maven-action/blob/9337884575c7c64d6ed20d9ab3ee943658d49336/action.yml#L112-L119) a caching step, which is [enabled by default](https://github.com/s4u/setup-maven-action/blob/9337884575c7c64d6ed20d9ab3ee943658d49336/action.yml#L50-L54).

Reply:
<img width="908" alt="Screenshot 2024-01-13 at 17 15 49" src="https://github.com/PicnicSupermarket/error-prone-support/assets/110535847/7252b46f-f76e-4a1a-a9a5-ac40e49f3de0">
Sorry I was being blind 😮‍💨